visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -35,41 +35,57 @@ export const ERROR_MESSAGES = {
 export function validate(nameDataPair: NameDataPair) {
 	const file = nameDataPair?.content
 	const result: CCValidationResult = { error: [], warning: [] }
+
+	result.error = checkErrors(file)
+	result.warning = checkWarnings(file)
+
+	if (result.error.length > 0) {
+		throw result.error
+	}
+}
+
+function checkWarnings(file: ExportCCFile) {
+	const warnings = []
+	switch (true) {
+		case fileHasHigherMinorVersion(file):
+			warnings.push(`${ERROR_MESSAGES.minorApiVersionOutdated} Found: ${file.apiVersion}`)
+			break
+	}
+	return warnings
+}
+
+function checkErrors(file: ExportCCFile) {
+	const errors = []
 	switch (true) {
 		case !file:
-			result.error.push(ERROR_MESSAGES.fileIsInvalid)
+			errors.push(ERROR_MESSAGES.fileIsInvalid)
 			break
 		case !isValidApiVersion(file):
-			result.error.push(ERROR_MESSAGES.apiVersionIsInvalid)
+			errors.push(ERROR_MESSAGES.apiVersionIsInvalid)
 			break
 		case fileHasHigherMajorVersion(file):
-			result.error.push(ERROR_MESSAGES.majorApiVersionIsOutdated)
-			break
-		case fileHasHigherMinorVersion(file):
-			result.warning.push(`${ERROR_MESSAGES.minorApiVersionOutdated} Found: ${file.apiVersion}`)
+			errors.push(ERROR_MESSAGES.majorApiVersionIsOutdated)
 			break
 	}
 
-	if (result.error.length === 0) {
+	if (errors.length === 0) {
```

Comment:
The JSON Schema validations is a separate concern and should therefore be extracted into a separate function.

Reply:
Implemented in the next commit.